commit d6696547e597703811213284a9b935224ff5fc1f
Author: Andy Boedo <andresboedo@gmail.com>
Date:   Wed Jul 24 11:28:45 2024 -0300

    add pricePerPeriod properties (#621)
    
    Added `pricePerWeek`, `pricePerMonth`, `pricePerYear` and string
    variants of them. Based on
    https://github.com/RevenueCat/purchases-ios/pull/3546
    
    Edits by @JayShortway:
    - Added variants for Android, iOS and Typescript.
    - This seems to be all that's needed for React
    Native and Kotlin Multiplatform, but I didn't check other hybrids yet.
    Please let me know if you see something that wont work in a certain
    hybrid.
    
    ---------
    
    Co-authored-by: JayShortway <29483617+JayShortway@users.noreply.github.com>

diff --git a/ios/PurchasesHybridCommon/PurchasesHybridCommon/StoreProduct+HybridAdditions.swift b/ios/PurchasesHybridCommon/PurchasesHybridCommon/StoreProduct+HybridAdditions.swift
index b839dbf0..67fcd107 100644
--- a/ios/PurchasesHybridCommon/PurchasesHybridCommon/StoreProduct+HybridAdditions.swift
+++ b/ios/PurchasesHybridCommon/PurchasesHybridCommon/StoreProduct+HybridAdditions.swift
@@ -10,6 +10,42 @@ import Foundation
 import RevenueCat
 import StoreKit
 
+@objc public extension StoreProduct {
+    
+    // Re-exports pricePerPeriod properties with different names to avoid recursion.
+
+    @available(iOS 11.2, macOS 10.13.2, tvOS 11.2, watchOS 6.2, *)
+    @objc var pricePerWeekAmount: NSDecimalNumber? {
+        return self.pricePerWeek
+    }
+
+    @available(iOS 11.2, macOS 10.13.2, tvOS 11.2, watchOS 6.2, *)
+    @objc var pricePerMonthAmount: NSDecimalNumber? {
+        return self.pricePerMonth
+    }
+
+    @available(iOS 11.2, macOS 10.13.2, tvOS 11.2, watchOS 6.2, *)
+    @objc var pricePerYearAmount: NSDecimalNumber? {
+        return self.pricePerYear
+    }
+
+    @available(iOS 11.2, macOS 10.13.2, tvOS 11.2, watchOS 6.2, *)
+    @objc var pricePerWeekString: String? {
+        return self.localizedPricePerWeek
+    }
+
+    @available(iOS 11.2, macOS 10.13.2, tvOS 11.2, watchOS 6.2, *)
+    @objc var pricePerMonthString: String? {
+        return self.localizedPricePerMonth
+    }
+
+    @available(iOS 11.2, macOS 10.13.2, tvOS 11.2, watchOS 6.2, *)
+    @objc var pricePerYearString: String? {
+        return self.localizedPricePerYear
+    }
+
+}
+
 internal extension StoreProduct {
 
     var rc_dictionary: [String: Any] {
@@ -21,27 +57,40 @@ internal extension StoreProduct {
             "introPrice": NSNull(),
             "price": self.price,
             "priceString": self.localizedPriceString,
+            "pricePerWeek": NSNull(),
+            "pricePerMonth": NSNull(),
+            "pricePerYear": NSNull(),
+            "pricePerWeekString": NSNull(),
+            "pricePerMonthString": NSNull(),
+            "pricePerYearString": NSNull(),
             "productCategory": self.productCategoryString,
             "productType": self.productTypeString,
             "title": self.localizedTitle,
             "subscriptionPeriod": NSNull(),
         ]
 
-        if #available(iOS 11.2, tvOS 11.2, macOS 10.13.2, *),
-           let introductoryDiscount = self.introductoryDiscount {
-            dictionary["introPrice"] = introductoryDiscount.rc_dictionary
-        }
+        if #available(iOS 11.2, tvOS 11.2, macOS 10.13.2, *) {
+            dictionary["pricePerWeek"] = self.pricePerWeek
+            dictionary["pricePerMonth"] = self.pricePerMonth
+            dictionary["pricePerYear"] = self.pricePerYear
+            dictionary["pricePerWeekString"] = self.localizedPricePerWeek
+            dictionary["pricePerMonthString"] = self.localizedPricePerMonth
+            dictionary["pricePerYearString"] = self.localizedPricePerYear
 
-        if #available(iOS 12.2, tvOS 12.2, macOS 10.14.4, *) {
-            dictionary["discounts"] = self.discounts.map { $0.rc_dictionary }
-        }
+            if let introductoryDiscount = self.introductoryDiscount {
+                dictionary["introPrice"] = introductoryDiscount.rc_dictionary
+            }
 
-        if #available(iOS 11.2, macOS 10.13.2, tvOS 11.2, watchOS 6.2, *) {
-            if let subscriptionPeriod = self.subscriptionPeriod {
-                dictionary["subscriptionPeriod"] = StoreProduct.rc_normalized(subscriptionPeriod: subscriptionPeriod)
+            if #available(iOS 12.2, tvOS 12.2, macOS 10.14.4, *) {
+                dictionary["discounts"] = self.discounts.map { $0.rc_dictionary }
             }
-        }
 
+            if #available(watchOS 6.2, *) {
+                if let subscriptionPeriod = self.subscriptionPeriod {
+                    dictionary["subscriptionPeriod"] = StoreProduct.rc_normalized(subscriptionPeriod: subscriptionPeriod)
+                }
+            }
+        }
         return dictionary
     }
 
